source/_posts/Planetscape-a-dystopian-escape-game-for-36C3.md

```diff
@@ -0,0 +1,189 @@
+
+![][img:lockpick]
+
+## Technology
+Planetscape components are:
+
+- A web dApp that players use to read the story, submit passwords, and check the leaderboard.
+- A smart contract to store the structure of the story, validate players' submissions, and read the leaderboard.
+- Encrypted JSON documents stored in IPFS that contain the actual text of the chapters.
+
+We pledged to create a *user-friendly application that would rely on decentralized technology only*. We want to live the decentralized dream with no compromises, developing a powerful dApp that would use Ethereum under the hood without exposing any details about blockchain and IPFS to the player. We committed to have **instant onboarding** by allowing people to play the game just by visiting a website.
+
+We achieved that by leveraging our learnings from [Hunt Zuckerberg][3] and [Planet A][4] by using a "burner-wallet"-like approach (the player's wallet is created on the fly when opening the dApp for the first time). In order to make transactions players needed Ether. Given that the game was played on the [Rinkeby][6] network, we set up a *gas station* that would simply [transfer][7] a small amount of Ether to new players.
+
+Leveraging blockchain technology and IPFS give us many advantages:
+
+- We don't have to manage any servers.
+- Each time a player finds a correct solution, a transaction is sent to the smart contract of the game. This allows us to easily create stats on the number of players and the difficulty of each level.
+- The smart contract of the game maps each chapter to its IPFS content hash, once the game is set up there is no way for the game masters to cheat.
+- (For the future) include other game masters to expand the main story or create sub-stories.
+
+### Introducing THC
+In order to orchestrate all different components of the game, Social Dist0rtion Protocol (yes, still us) developed a game engine called [Treasure Hunt Creator][0] (**THC** for short) that, given a story structured in directories, is able to:
+
+- Build and deploy the front-end application.
+- Build, deploy and initialize the game smart contract on the Ethereum Network.
+- Encrypt and deploy the chapters on IPFS.
+
+[THC][0] played a crucial role in the creation of the game, allowing us to meet our high standards in terms of development experience. But THC was more than that. Fast prototyping and automatic deploying gave us the piece of mind we needed during our last hours of work.
+
+### Zero Knowledge Proof
+As mentioned earlier, every time a correct password is found, a transaction is sent to the smart contract managing the game. Ethereum transactions are public, and we don't want players to cheat sniffing passwords submitted by others. To avoid that, we implemented a cheap zero-knowledge proof scheme. Roughly speaking, the scheme is the following:
```

Comment:
```suggestion
As mentioned earlier, every time a correct password is found, a transaction is sent to the smart contract. Ethereum transactions are public and we don't want players to sniff passwords submitted by others. To avoid that, we implemented a cheap zero-knowledge proof scheme. Roughly speaking, the scheme is the following:
```